### DISCUSSION TITLE
[FEATURE REQUEST] Rearranging workspaces

### DISCUSSION THREAD

**Discussion by edofe99** [linuxmint/discussions#495] [Ideas]
Can you please add the possibility of rearranging the order of workspaces? Thank's.

**Comment by klangman** [comment]
How about just swapping all the windows between two workspaces. Same effect really. This could easily be done with a new applet.

**Reply by rsv-code** [reply]
This would work if you only worked off of two workspaces and only had two windows open. Many of us work off of 4 or more with many windows open, and the order matters. So if we need to create a new workspace and insert it in the middle of existing workspaces, this means manually moving down all the windows in all the workspaces below where you need it. It becomes hard to manage at that point.

**Comment by rsv-code** [comment]
+1 I've been a Gnome user for a very long time and recently switched my laptop over to LinuxMint Cinnamon. I'm really impressed and love everything about it, but this missing feature is a major pain point for me. I use it so much so this may be a deal breaker for me. Thanks

**Comment by Jopp-gh** [comment]
Luckily rearranging workspaces isn't very important for me, still I'm wondering why I've to rename workspaces to arrange them. 

Shouldn't be a big deal to change the index of available workspaces by drag and drop.

**Comment by klangman** [comment]
OK.. I'll make an applet for this. It shouldn't take too long (I hope) for stage one anyhow (hotkeys that let you swap adjacent workspace). Drag-and-Drop might take a little longer.

**Comment by klangman** [comment]
Here is the initial version on my [Github](https://github.com/klangman/RearrangeWorkspaces). I'll push to Cinnamon Spices after some more testing.

**Comment by klangman** [comment]
It's in Cinnamon Spices now: https://cinnamon-spices.linuxmint.com/extensions/view/112
